src/LoggyCore.hs

```diff
@@ -14,10 +14,14 @@ extractTimestamp dformat line = parsedTs
             _ -> error "Invalid log line!"
         timeParser = readSTime True defaultTimeLocale dformat  
 
-mergeLogLines :: DateFormat -> [LogLine] -> [LogLine] -> [LogLine]
-mergeLogLines dformat l1 l2 = sortedLogLines
+mergeLogLines :: DateFormat -> [[LogLine]] -> [LogLine]
+mergeLogLines dformat lstLines = mergeLogLines' dFormatPerFile lstLines
+    where dFormatPerFile = replicate (length lstLines) dformat
+
+mergeLogLines' :: [DateFormat] -> [[LogLine]] -> [LogLine]
+mergeLogLines' dformats lstLines = sortedLogLines
     where
         sortedLogLines = snd <$> sortedLogTuples
-        sortedLogTuples = sortWith fst $ zip logTimestamps l3
-        logTimestamps = extractTimestamp dformat <$> l3
-        l3 = l1 ++ l2
\ No newline at end of file
+        sortedLogTuples = sortWith fst $ zip logTimestamps ls
+        logTimestamps = concat $ zipWith ($) (fmap.extractTimestamp <$> dformats) lstLines
+        ls = concat lstLines
```

Comment:
```suggestion
sortedLogTuples = sortWith fst tsAndLines
tsAndLines = (\fmt l -> (extractTimestamp fmt l, l) =<< zip dformats lstLines
```

untested 🙈

Reply:
No, it doesnt seem to be working.
Say we have the inputs, `["df1", "df2"]` and `[["l1", "l2"], ["l3"]]`
`zip dformats lstLines` == `[("df1",["l1","l2"]),("df2",["l3"])]`
When applying `(\fmt l -> (extractTimestamp fmt l, l)`, it will try to apply extractTimestamp to 1 `DateFormat` and a `[[LogLine]]`.